web/modules/custom/error_page/src/ErrorPageErrorHandler.php

```diff
@@ -0,0 +1,311 @@
+        '%file' => $caller['file'],
+        '%line' => $caller['line'],
+        'severity_level' => $severity_level,
+        'backtrace' => $backtrace,
+        '@backtrace_string' => (new \Exception())->getTraceAsString(),
+      ], $recoverable || $to_string);
+    }
+    // If the site is a test site then fail for user deprecations so they can be
+    // caught by the deprecation error handler.
+    elseif (DRUPAL_TEST_IN_CHILD_SITE && $level === E_USER_DEPRECATED) {
+      $backtrace = debug_backtrace();
+      $caller = Error::getLastCaller($backtrace);
+      _drupal_error_header(
+        Markup::create(Xss::filterAdmin($message)),
+        'User deprecated function',
+        $caller['function'],
+        $caller['file'],
+        $caller['line']
+      );
+    }
+  }
+
+  /**
+   * Handles exceptions.
+   *
+   * Code copied and adapted from _drupal_exception_handler().
+   *
+   * @param \Throwable $exception
+   *   The throwable.
+   *
+   * @see _drupal_exception_handler()
+   */
+  public static function handleException(\Throwable $exception) {
+    require_once DRUPAL_ROOT . '/core/includes/errors.inc';
+
+    try {
+      // Log the message to the watchdog and return an error page to the user.
+      static::logError(Error::decodeException($exception), TRUE, $exception);
+    }
+    // PHP 7 introduces Throwable, which covers both Error and
+    // Exception throwables.
+    catch (\Throwable $error) {
+      _drupal_exception_handler_additional($exception, $error);
+    }
+    // In order to be compatible with PHP 5 we also catch regular Exceptions.
+    catch (\Exception $exception2) {
+      _drupal_exception_handler_additional($exception, $exception2);
+    }
+  }
+
+  /**
+   * Logs a PHP error or exception and displays an error page in fatal cases.
+   *
+   * This method is a slight changed version of _drupal_log_error().
+   *
+   * @param array $error
+   *   An array with the following keys: %type, @message, %function, %file,
+   *   %line, @backtrace_string, severity_level, and backtrace. All the
+   *   parameters are plain-text, with the exception of @message, which needs to
+   *   be an HTMLstring, and backtrace, which is a standard PHP backtrace.
+   * @param bool $fatal
+   *   TRUE for:
+   *   - An exception is thrown and not caught by something else.
+   *   - A recoverable fatal error, which is a fatal error.
+   *   Non-recoverable fatal errors cannot be logged by Drupal.
+   * @param \Throwable|null $original_exception
+   *   The original exception.
+   *
+   * @see _drupal_log_error()
+   */
+  protected static function logError(array $error, $fatal = FALSE, $original_exception = NULL) {
+    $settings = Settings::get('error_page');
+    $uuid_enabled = isset($settings['uuid']['enabled']) ? $settings['uuid']['enabled'] : TRUE;
+    static::$uuidAddToMessage = $uuid_enabled && (isset($settings['uuid']['add_to_message']) ? $settings['uuid']['add_to_message'] : TRUE);
```

Comment:
This is used only in the `::getMessage()` function, we should move it there. But on the big picture, we should remove this functionality.